src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -220,7 +222,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             sym::discriminant_value => {
                 let place = self.deref_operand(args[0])?;
                 let discr_val = self.read_discriminant(place.into())?.0;
-                self.write_scalar(Scalar::from_uint(discr_val, dest.layout.size), dest)?;
+                self.write_scalar(Scalar::from_u64(u64::try_from(discr_val).unwrap()), dest)?;
```

Comment:
@eddyb this is where Miri would now ICE for too big discriminants. (And I am not entirely sure what happens for negative discriminants, TBH...)

Reply:
Looking at [this](https://github.com/rust-lang/rust/blob/e768d6f0bc6db7a46c9ef08254a944ba100bc5dd/src/librustc_mir/interpret/operand.rs#L635), I think for signed 64bit discriminants we appropriately truncate and everything is okay. (I tried to find a way to share that code there with `cast.rs` but found no good way.)

For negative discriminants with `repr(i128)` this will ICE, but well, these just cannot be represented properly in a `u64`...